apps/prairielearn/src/components/PersonalNotesPanel.html.ts

```diff
@@ -0,0 +1,244 @@
+import { escapeHtml, html } from '@prairielearn/html';
+
+import { config } from '../lib/config.js';
+import type { AssessmentInstance, File } from '../lib/db-types.js';
+
+export function PersonalNotesPanel({
+  fileList,
+  courseInstanceId,
+  assessment_instance,
+  authz_result,
+  variantId,
+  allowNewUploads = true,
+  csrfToken,
+}: {
+  fileList: File[];
+  courseInstanceId: string;
+  assessment_instance: Pick<AssessmentInstance, 'id' | 'open'>;
+  authz_result: Record<string, any>;
+  variantId?: string;
+  allowNewUploads?: boolean;
+  csrfToken: string;
+}) {
+  return html`
+    <div class="card mb-4" id="attach-file-panel">
+      <div class="card-header bg-secondary text-white">
+        <i class="fas fa-paperclip"></i>
+        Personal Notes
+      </div>
+      <table class="table table-sm">
+        <tbody>
+          ${fileList.length === 0
+            ? html`
+                <tr>
+                  <td><i>No attached notes</i></td>
+                </tr>
+              `
+            : fileList.map(
+                (file) => html`
+                  <tr>
+                    <td style="word-break:break-all;">
+                      <a
+                        class="attached-file"
+                        href="${config.urlPrefix}/course_instance/${courseInstanceId}/assessment_instance/${assessment_instance.id}/file/${file.id}/${file.display_filename}"
+                      >
+                        ${file.display_filename}
+                      </a>
+                    </td>
+                    ${assessment_instance.open &&
+                    authz_result.active &&
+                    authz_result.authorized_edit &&
+                    allowNewUploads
+                      ? html`
+                          <td style="width:1%; text-align:right;">
+                            ${file.type === 'student_upload'
+                              ? DeletePersonalNoteButton({ file, variantId, csrfToken })
+                              : ''}
+                          </td>
+                        `
+                      : ''}
+                  </tr>
+                `,
+              )}
+        </tbody>
+      </table>
+
+      ${allowNewUploads
+        ? html`
+            <div class="card-footer">
+              ${!assessment_instance.open || !authz_result.active
+                ? html`
+                    <p class="small mb-0">
+                      Notes can't be added or deleted because the assessment is closed.
+                    </p>
+                  `
+                : !authz_result.authorized_edit
+                  ? html`
+                      <div class="alert alert-warning mt-2" role="alert">
+                        You are viewing the
+                        ${variantId != null ? 'question instance' : 'assessment instance'} of a
```

Comment:
This has been changed from checking for `question` to checking for `variantId`, since in all contexts where this panel is called, if there is a question, there is a variant. This allowed the component to avoid a parameter that was used solely for this check.